### DISCUSSION TITLE
Passing "Channel Behavior Analysis"

### DISCUSSION THREAD

**Discussion by samadtehrani** [rrirower/DP2SceneGraphWizard#38] [Q&A]
Hi,
Thank you @rrirower  for the great effort here and for the DP community.
I did used the code and the my channel works good so far as a side-loaded app. (Alpha-6 at this moment) 
So I tried to upload the package to my Roku account, Followed the instruction, and everything went smoothly till I hit the bump of "Channel Behavior Analysis" 
Looks like the "Channel Behavior Analysis" has a lot of issues. for example the Deep Linking section scripts does not work and the actual Content ID is not playing at all. Also Play performance is not play anything and show the Error instead. 
I think I read somewhere that @ferdiworks published some of his channels. So I wanted to ask if anyone here passed this step?  
I know this discussion is not related to software here and I will ask the question on Roku forum upon admin's decision
![Untitled](https://github.com/rrirower/DP2SceneGraphWizard/assets/139641942/bc504c57-6343-4307-b459-b067517f8443)

**Comment by rrirower** [comment]
@samadtehrani Can you send me the text of the errors listed in your image?  Or, any other info that explains what the problem is.

I'm pretty sure that other users have successfully passed these criteria.  Let's see if anyone else responds.

**Reply by rrirower** [reply]
@samadtehrani One other thing.... Version 1.0.0.0-alpha-6 was reposted after its initial post. You should make sure you have the correct version, or, use the version I just posted.

**Comment by samadtehrani** [comment]
Thank you for your reply. 
Yes, I used the latest version of alpha-6. (I did try the alpha-7 but I rather to stay safe with the old working one yet)
Somehow the action column has general info, the script they are using with the Remote software on that specific Roku device, and a screenshot of the device on that stage. I tried to attach the scrip file but the github does not allow for security reasons. 

It does not explain anything about my channel test and what happened during the test. 
When I use the scrip and use it on the Roku's remote software and my side-loaded channel I get this errors. 
![Channel content play performance](https://github.com/rrirower/DP2SceneGraphWizard/assets/139641942/7160c8d4-a43c-410d-a34d-aa992ae7749d)
![Channel Deep Linking Basic](https://github.com/rrirower/DP2SceneGraphWizard/assets/139641942/b9859efc-598a-4156-b11a-c7c744254c29)
![Channel Launch Performance](https://github.com/rrirower/DP2SceneGraphWizard/assets/139641942/45e4eb8a-9da2-4d4e-a02c-eef3158c7389)
![Focused Full-Screen Video Playback](https://github.com/rrirower/DP2SceneGraphWizard/assets/139641942/e330b4f9-1332-45e7-8ad0-feaad361d241)

I think I will wait for the others  to share their successful experience.

**Reply by rrirower** [reply]
One thing that jumps out at me is the "special" media type.  I'm not sure that's valid.  Can you post your feed file so I can have a look?

**Comment by samadtehrani** [comment]
![Untitled](https://github.com/rrirower/DP2SceneGraphWizard/assets/139641942/aafe8b5d-9b7d-47d2-9bf4-9e36fa0908c5)
You are absolutely right. It should be "tvSpecials" as it is in my feed. ([Attached](http://207.246.125.227/melo_feed_002.json)) . In deeplinking section (dev account), there is only "special" not  "tvSpecials". 

So I did change it shortFormVideos in my feed and deeplink section (dev account), and unfortunately, I got the same result.
But I think I found the problem:
"Channel Behavior Analysis Set up" is in your task list if you select monetization. and that's why @ferdiworks or others did publish non-monetized channels. So I switched to alpha-7... and I got the same result.

**Reply by rrirower** [reply]
Are you running the channel static analysis tool standalone or from the website?

**Reply by samadtehrani** [reply]
From the website

**Reply by rrirower** [reply]
I just realized that you posted from the Roku remote tool. Did you invoke it directly, or, did the online analysis tool start it?  I'm a bit confused because you reported on the channel analysis but your images are of something else.

**Reply by samadtehrani** [reply]
As I mentioned above, the page "Channel Behavior Analysis" gives us some scripts. They use these scripts with the "Roku Remote Tool" to analyze the Deep linking and other stuff. I tried to reproduce the Errors on my side-loaded channel with the same script and "Roku Remote Tool".  and the screenshots were the results.

**Reply by ferdiworks** [reply]
@samadtehrani I published many channels with TV Specials, select specials and deep link a video. Ensure your tvspecials have cateogory tags and playlists! all should pass!

**Reply by samadtehrani** [reply]
Thank you @ferdiworks Did you pass "Channel Behavior Analysis" in your published channel? or "Static Analysis"? 
I assume your channels do not have ads so probably do not encounter that section of the checklist. 
Please try to select "My channel contains ads." in the Monetization section. 
![2](https://github.com/rrirower/DP2SceneGraphWizard/assets/139641942/73cc761f-8489-4ef3-b4bc-239d537020d0)

**Reply by ferdiworks** [reply]
@samadtehrani I do not have Ads! I past everything with no issues on about 35 channels converted with the Alpha 5 and Alpha 6 version. Then I use my own templates for other channels. This software is for simple channels!

**Reply by samadtehrani** [reply]
Since Alpha-7 ads are supported.

**Comment by samadtehrani** [comment]
Has anyone tried DP2SceneGraph Wizard with ads and successfully passed "Channel Behavior Analysis"?

**Reply by rrirower** [reply]
I can't run the Channel Behavior Analysis.  I don't currently have a channel that uses RAF.  Have you tried contacting Partner Success for help?  If you can give me more detailed information, I can try to look into the code, but, my hands are somewhat tied since I don't have access to the tests like you have.

**Comment by adamesq** [comment]
Curious about this, since I did want to try adding ads. Is it possible to create a test ad feed with a couple of test ads? I ran across a Roku one, but the entire content of the feed is:
`<Playlist/>
`

**Reply by rrirower** [reply]
I've got a Roku ad server url I've been using for testing. I'll post it later.

**Reply by adamesq** [reply]
🙏🏻

**Reply by rrirower** [reply]
Here is the Roku Ad server URL I've been using for testing:

[https://devtools.web.roku.com/samples/sample.xml](https://devtools.web.roku.com/samples/sample.xml)

**Comment by ferdiworks** [comment]
Has anyone used V8 and successfully published a channel with Ads?

**Comment by dhacker29** [comment]
I am at a loss on what to enter here so that it will pass the Channel Behavior Analysis. Any help would be appreciated.
![image](https://github.com/rrirower/DP2SceneGraphWizard/assets/981910/4f7a90ba-8f76-415e-9c77-7d9f95aca529)

**Reply by rrirower** [reply]
@ferdiworks can you help @dhacker29 ?

**Reply by ferdiworks** [reply]
From dropdown select video type from your feed file, example = shortform or movie or tvspecial or series!
Content ID = Add one video ID number from your feed file
Content Title = Add The video title!
Only do this for one video click to save it = Done!
Publish channel!

**Reply by dhacker29** [reply]
Thank you! Tests all passed and ready to publish.

**Comment by jasontalbott** [comment]
I'm wondering what I did wrong. My json feed is here: https://roku.tomorrowsworld.org/tomorrowsworld/feed.json and I used the Channel Feed Manager tool to check the feed file for any errors... no errors, no warnings. It is a pretty simple channel, no ads, no user accounts, no frills. Generated a package using the Wizard (version 1.0.0.2) It works great when I sideload it to my device but I am having issues when trying to submit it to ROKU. It passes Static Analysis although it shows warnings:

- [Billing usage was found in BrightScript code but was not specified on the Monetization channel properties page.], 
- [For your channel to pass certification, your application must fire the "AppDialogInitiate" and "AppDialogComplete" beacons if the channel UI displays a login, user selection, EULA, or any other dialog before the home page.], 
- [RAF usage was found in BrightScript code but Ads revenue was not specified during channel publishing.], 
- [We have found a proper integration of an Automatic Account Link in your channel code, but you have not selected Customer Account Requirement in channel submission properties via Developer Portal.]

I get a green checkmark for Static Analysis despite those warnings. However I have problems in the Channel Behavior Analysis. Error on all 6 tests. 
1. Channel Launch Performance - Error, Requirement 3.2 - Channels must launch to a fully rendered home screen within 15 seconds. A signal beacon must be added to the channel application to measure launch times. The debug console can then be used to verify that the channel's launch time is meeting this requirement. See Measuring channel performance on how to measure channel launch times.
2. Channel Deep Linking Basic - Error, Requirement 5.1 - Channels must support deep linking for all media types, per Roku's deep linking policy. Live streams (and replays of live broadcast streams) may be exempt from this requirement.
3. (Same as 2, I guess because I have two media types episodes and movies)
4. Channel Content Play Performance - Error, Requirement 3.6 - Channels must start playing content within 8 seconds of initiation.
5. Focused Full-Screen Video Playback - Error/Warning, Requirement 4.4 - Channels must keep focus on the Video node during full-screen video playback when there are no active UI components over the video. Roku reserves use of the Options ("*") button when in this state for the sole purpose of Roku system displays. Channels may use the Options button for additional in-channel options while UI components are displayed on top of the video.
6. Screensaver Policy - Error/Warning, Requirement 4.5 - Channels are prohibited from overriding or interfering with Roku's system screensaver. See Roku’s Screensaver Policy.

The only modification to the code I generated from the wizard was add the necessary fonts to the fonts folder and updated the theme.json accordingly. Any ideas?  FWIW, the sideloaded channel is great, it just would be nice to share it with the public.

**Reply by rrirower** [reply]
I won't be able to take a close look at this until Thursday. But, on initial inspection, it looks like you're not using a CDN for your content. That may contribute to performance issues. 

Perhaps someone else in this discussion can help.

**Reply by jasontalbott** [reply]
Not sure that's the problem, although thank you for pointing that out. I was under the impression for years that we had CloudFlare caching our video content and when you said that, I checked with our admin found that was not the case. She just turned it on so maybe that will help with the occasional buffering we've (rarely) seen.

**Comment by samadtehrani** [comment]
Looks like Roku's Behavioral Analysis system is broken again. I tried to pass another channel today and encountered the same problem as @jasontalbott 
in the past it took some time (5 to 10 minutes) to test the channel, in my experience, right now after just a few seconds all the fields are complete and there is no message for the error. please check the attached screenshots
![1](https://github.com/rrirower/DP2SceneGraphWizard/assets/139641942/2d0825a8-6d74-4caf-9651-ec4fe12f034c)
![3](https://github.com/rrirower/DP2SceneGraphWizard/assets/139641942/403985eb-5a60-4a99-bb91-9eaced35ffe8)

**Reply by jasontalbott** [reply]
Thanks, that's somewhat relieving to hear, maybe there's nothing wrong with my package then. I thought the "unknown error" messages were odd.

**Reply by jasontalbott** [reply]
Funny because I attended the webinar Roku had yesterday for the resource monitor 3.0 rollout and was testing my channel and thinking "wow, this runs really well!" ...but not according to Behavioral Analysis apparently.

**Reply by jasontalbott** [reply]
@samadtehrani if you are interested "tom_from_roku" is investigating and is asking if anyone else is experiencing the issue in a thread on the Roku developer Slack channel.  https://rokudevelopers.slack.com/archives/CD88K1TUJ/p1716306544190109

**Reply by samadtehrani** [reply]
Thanks. I will reply to that.

**Reply by samadtehrani** [reply]
It's on the Roku forum too:
https://community.roku.com/t5/Roku-Developer-Program/Channel-behavior-analysis/m-p/969960#M53473

**Reply by rrirower** [reply]
Thanks for following up on this. Until I know otherwise, I'll assume this is not caused by the wizard code.

**Reply by jasontalbott** [reply]
"tom_from_roku" told me via Slack that they've found the root cause and are actively working on a fix.

**Reply by jasontalbott** [reply]
It's working again. Resubmitted exact same package and it passed this time. My channel should be updated by Monday and I'll be happy to share in success stories.
